e2etests/all_test.go

```diff
@@ -0,0 +1,135 @@
+// +build e2etests
+
+package e2etests
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/gobuffalo/envy"
+	"github.com/stretchr/testify/suite"
+)
+
+type E2eSuite struct {
+	suite.Suite
+	goBinaryPath   string
+	env            []string
+	tmpDir         string
+	goModCache     string
+	sampleRepoPath string
+}
+
+type catalogRes struct {
+	Modules []struct {
+		Module  string `json:"module"`
+		Version string `json:"version"`
+	} `json:"modules"`
+}
+
+func (m *E2eSuite) SetupSuite() {
+	var err error
+	m.tmpDir, err = mkTemp()
+	if err != nil {
+		m.Fail("Failed to make temp dir", err)
+	}
+
+	m.goModCache = path.Join(m.tmpDir, "pkg", "mod")
```

Comment:
I would probably make two temp directories to maintain a nicer separation: one for GOPATH and one outside of GOPATH for happy-path. This way you don't to actually care about the "pkg/mod" part here, and cleaning the cache only needs to take the GOPATH env var and nothing else

Reply:
Done